atoma-proxy/src/server/mod.rs

```diff
@@ -278,36 +386,40 @@ pub struct NodePublicAddressAssignment {
     public_address: String,
 }
 
-async fn models_handler() -> Result<Json<Value>, StatusCode> {
+async fn models_handler(State(state): State<ProxyState>) -> Result<Json<Value>, StatusCode> {
     Ok(Json(json!({
         "object": "list",
-        "data": [
-          {
-            "id": "meta-llama/Llama-3.2-3B-Instruct",
-            "object": "model",
-            "created": 1730930595,
-            "owned_by": "vllm",
-            "root": "meta-llama/Llama-3.2-3B-Instruct",
-            "parent": null,
-            "max_model_len": 2048,
-            "permission": [
-              {
-                "id": "modelperm-f3fdf969d5b54544ba451b1f5785325a",
-                "object": "model_permission",
-                "created": 1730930595,
-                "allow_create_engine": false,
-                "allow_sampling": true,
-                "allow_logprobs": true,
-                "allow_search_indices": false,
-                "allow_view": true,
-                "allow_fine_tuning": false,
-                "organization": "*",
-                "group": null,
-                "is_blocking": false
-              }
-            ]
-          }
-        ]
+        "data": state
+        .models
+        .iter()
+        .map(|model| {
+            json!({
+              "id": model,
+              "object": "model",
+              "created": 1730930595,
+              "owned_by": "vllm",
+              "root": model,
+              "parent": null,
+              "max_model_len": 2048,
+              "permission": [
+                {
+                  "id": format!("modelperm-{}", model),
```

Comment:
Where is the metadata below coming from ?

Reply:
This is what vllm returned and I just used it. I have to check what this structure should/can.
Added todo.